### DISCUSSION TITLE
When to upgrade the kernel?

### DISCUSSION THREAD

**Discussion by Rom888** [AdityaGarg8/mbp-16.x-ubuntu-kernel#51] [Q&A] [answered]
Sometimes I receive notifications about a new kernel being released.
Could you please explain if I should always upgrade it as soon as possible?

I have:
Ubuntu 22.04
6.5.5-t2-jammy
macbook pro, 16 inch, 2019

**Comment by AdityaGarg8** [comment, marked as answer]
Its not needed always. But sometimes new features are added (like newer driver for touchbar in 6.5.7-2) and bugs are fixed (Eg: screen resolution bug fixed in 6.5.6-1).

Its recommended that you upgrade your kernel every 2 weeks imo.

**Reply by Rom888** [reply]
Thank you!

**Comment by AdityaGarg8** [comment]
In case you upgrade your kernel to 6.5.7-2 or above , after upgrading install `tiny-dfr` and restart to use the new touchbar driver.

Run `sudo touchbar` to configure your touchbar.
